docs/build-results-archive.md

```diff
@@ -0,0 +1,52 @@
+# Build results archive.
+
+This document describes a build results archive format which can be used to store the sarif results of a build-and-test procedure.
+
+The archive will take the form of a compressed (zstd if available, or bzip2 otherwise) tar file containing the following files.
+
+```
+./build-results-archive
+./build/**/*.sarif
+./colcon-build-cmd
+./colcon-test-cmd
+./processed.sarif
+./vcs-export-exact.repos
+```
+
+The individual files are further described:
+
+## build-results-archive
+
+This is a JSON file containing metadata regarding the current archive.
+The file _must_ contain an single top-level JSON object and that object _must_ contain the `"version"` key with the value of `1`.
+Future revisions of this format may add additional valid versions.
+
+## build/**/*.sarif
+
+The build/ subdirectory _must_ contain the unaltered sarif content of the workspace build directory after the end of the latest colcon test run.
```

Comment:
Should we also state that the files should be read-only, just to reinforce the idea that they are immutable?

Reply:
I think that the archive itself should be entirely read-only and immutable. It's worth noting that in the text but whether the files themselves are "read-only" or not will be implementation dependent. (Ours will probably try to make them as such)

Reply:
Noted in https://github.com/space-ros/process_sarif/pull/6/commits/b4440873d32415b983db5dbf23889cdfe06c9108 along with a suggestion that applications using the data use their own formats rather than alter these files.